utxonursery.go

```diff
@@ -281,17 +281,28 @@ type incubationRequest struct {
 // defined within the summary of a closed channel. Individually, as all outputs
 // reach maturity they'll be swept back into the wallet.
 func (u *utxoNursery) incubateOutputs(closeSummary *lnwallet.ForceCloseSummary) {
-	outputAmt := btcutil.Amount(closeSummary.SelfOutputSignDesc.Output.Value)
-	selfOutput := &kidOutput{
-		amt:              outputAmt,
-		outPoint:         closeSummary.SelfOutpoint,
-		blocksToMaturity: closeSummary.SelfOutputMaturity,
-		signDescriptor:   closeSummary.SelfOutputSignDesc,
-		witnessType:      commitmentTimeLock,
+	var incReq *incubationRequest
+
+	// It could be that our to-self output was below the dust limit. In that
+	// case the SignDescriptor would be nil and we would not have that output
+	// to incubate.
+	if closeSummary.SelfOutputSignDesc != nil {
+		outputAmt := btcutil.Amount(closeSummary.SelfOutputSignDesc.Output.Value)
+		selfOutput := &kidOutput{
+			amt:              outputAmt,
+			outPoint:         closeSummary.SelfOutpoint,
+			blocksToMaturity: closeSummary.SelfOutputMaturity,
+			signDescriptor:   closeSummary.SelfOutputSignDesc,
+			witnessType:      commitmentTimeLock,
+		}
+
+		incReq.outputs = append(incReq.outputs, selfOutput)
```

Comment:
This line is triggering a panic in the integration tests. The `incReq` struct needs to be properly initialized:
```
var inqReq incubationRequest
```

Then:
```
u.requests <- &incReq
```